### DISCUSSION TITLE
How to do a horizontal line in a vertical flowchart without the lines being all messed up?

### DISCUSSION THREAD

**Discussion by chiafish** [mermaid-js/mermaid#5548] [Q&A]
Haven't seen anyone with this issue yet and it's driving me nutssssss
My knowledge on mermaid is pretty limited and I just started learning how it works since I want to use it in Notion.
This is what I want:
<img width="437" alt="Skärmavbild 2024-05-28 kl  15 00 26" src="https://github.com/mermaid-js/mermaid/assets/126991555/11c79449-e4ff-4fcc-95fa-1f61d0c87e25">

This is the best I can do lmao:
<img width="464" alt="Skärmavbild 2024-05-28 kl  14 58 07" src="https://github.com/mermaid-js/mermaid/assets/126991555/4da6a7ba-b2c3-4de1-b0e8-2427ee7b87b0">

Any suggestions?

**Comment by PstasDev** [comment]
Hi there!

Mermaid syntax can indeed be a bit tricky when you're trying to do something out of the ordinary, like mixing flow directions.

Here's a snippet that might help you achieve a layout close to what you're describing:

```mermaid
flowchart TB
    subgraph main
    A --- B --- C
    end
    subgraph for_rightbox
        direction RL
        B --- Rightbox
    end
    subgraph for_leftbox
        direction LR
        Leftbox --- C
    end
```

This setup uses subgraphs to change the direction of the flow for specific parts of your chart. While Mermaid doesn't offer extensive options for relative or absolute positioning of boxes directly in its syntax, this approach might be the closest to achieving a mixed-direction flowchart.

If you're looking for more advanced layout options, I recommend checking under the Directives menu of the Mermaid documentation: [Mermaid Directives](https://mermaid.js.org/config/directives.html). There might be additional configurations there that could help fine-tune your flowchart's appearance.

Hope this helps, and happy diagramming!
